fdbserver/MasterProxyServer.actor.cpp

```diff
@@ -808,10 +808,11 @@ ACTOR Future<Void> releaseResolvingAfter(ProxyCommitData* self, Future<Void> rel
 // Commit one batch of transactions trs
 ACTOR Future<Void> commitBatch(
 	ProxyCommitData* self,
-	vector<CommitTransactionRequest> trs,
+	vector<CommitTransactionRequest>* pTrs,
 	int currentBatchMemBytesCount)
 {
 	//WARNING: this code is run at a high priority (until the first delay(0)), so it needs to do as little work as possible
+	state std::vector<CommitTransactionRequest> trs(std::move(*(const_cast<std::vector<CommitTransactionRequest>*>(pTrs))));
```

Comment:
Why do we need a const_cast here?

Reply:
The reason is that the flow generated constructor is typing `trs` `std::vector<CommitTransactionRequest> const*`, the constness has to be casted off.